### PR TITLE
AB#362 bug

### PR DESCRIPTION
**Pull request by camagallen-exac**
## Overview
Corrects package compatibility issues and consequently corrects two problems that were preventing development of this fork: attachment to KenSci resources and a bug in url testing. The Docker image is now updated to use a public base image. The bug in URL testing was revealed by the new Docker image and corrected to enable all tests to pass as expected.

Resolves [AB#362](https://dev.azure.com/exactech/3f1b3419-4ffb-4ebb-804f-ae494866bc10/_workitems/edit/362)

## How was this update tested?
All pytests passed, both locally and using the updated Docker image.

## Checklist
- [x] I have read the [Contributing](https://github.com/KenSciResearch/docs/code_contributions/CONTRIBUTING.md) guidelines
- [x] I performed a self-review of my own code
- [x] The CHANGELOG and version number (setup.cfg) have been updated
- [x] Documentation has been updated to reflect the changes
    - [x] The code has clear and helpful comments, particularly in hard-to-understand areas
    - [x] READMEs, Tutorials/Examples, and links have been updated as appropriate
- [x] Tests have been added to prove that the fix is effective or that the feature works
- [x] Pytest was run; new and existing unit tests pass locally with my changes
- [x] I have made note of any breakages or dependencies (if applicable)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] These changes generate no new warnings

### CODE REVIEW

**Review by camagallen-exac — COMMENTED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file. Please do f
 
 _“Yesterday I was clever, so I wanted to change the world. Today I am wise, so I am changing myself.”_ - Rumi
 
+## [1.1.0] - 2023-07-24
+### Installation and Test Corrections
+- Updated Docker image to enable development outside of KenSci
```
Comment (kenviku) [inline CHANGELOG.md]: nit: `updated docker to use publicly available python image - python:3.9-bookworm`? 
(helpful link: https://hub.docker.com/_/python)

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
-FROM kentoso.azurecr.io/kensciml/python37:latest AS setup
+FROM python:3.9-bookworm
 
 RUN apt-get update && apt-get install -y libgomp1
 RUN apt-get install -y libgomp1 gcc g++
 
-WORKDIR /kensci
+WORKDIR /fmlhealth
```
Comment (kenviku) [inline Dockerfile]: nit: use consistent naming - `fairmlhealth`?. If there is a directory inside working directory already named `fairmlhealth` (for the name of the package) then you can choose "dev" or may be "evaluation" (so as to reflect that this repository is specific to evaluation).

File: test/__utils.py [hunk 3]
```diff
@@ -34,16 +34,15 @@ def get_urls(text_string):
...
-            raw_urls[1] = "http" + raw_urls[1]
-            continue
-        else:
-            # Remove errant leading & trailing symbols will be recognized as
-            #   part of the url
-            url = url.rstrip()
-            while __invalid_url_delimiter(url[-1]):
-                url = url[:-1]
-            output.append(url)
+            temp_url = url.split("](http")
```
Comment (kenviku) [inline test/__utils.py]: Can the regex above be updated to overcome this issue with markdown format?

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file. Please do f
...
+- Corrected bug in url tester that was revealed by the new docker image (required)
```
Comment (kenviku) [inline CHANGELOG.md]: It is not clear what was the bug. May be a bit more specific?

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file. Please do f
...
+- Corrected recent package incompatibility that blocked installation
```
Comment (kenviku) [inline CHANGELOG.md]: nit: being specific is recommended. What was the blocker? which package?

**Review by kenviku — CHANGES_REQUESTED** [review]
Overall good. Some nits and questions.

File: test/__utils.py (hunk 3 above)
Comment (camagallen-exac) [reply test/__utils.py]: I don't have enough experience with regex to come up with something in a reasonable amount of time. Will add this as a bug to address as time allows.

Comment (camagallen-exac) [reply test/__utils.py]: Issue resolved offline.

**Comment by camagallen-exac** [comment]
Code and markdown files updated to better validate URLs

**Review by kenviku — APPROVED** [review]
LGTM. Thanks for adding the regex edits.

**Pull request merged**